src/java/org/apache/cassandra/db/marshal/UserType.java

```diff
@@ -67,7 +68,10 @@ public UserType(String keyspace, ByteBuffer name, List<FieldIdentifier> fieldNam
         {
```

Comment:
Is there any reason this loop isn't just `for (int i = 0; i < fieldNames.size(); i++)`?

Reply:
I think because its dealing with a map and not a list, so converts to an entity set.  Could use an iterator to get similar results.

Reply:
An old habit :-) In some previous life some collection `size()` methods were costly to call so I took the habit to call the size method only once in my loops. Feel free to change it. To be honest, I have probably done that with every `for` loop I wrote in C*.

Reply:
It's good practice to hoist the size, in case the JVM cannot determine the specific type of the collection/list

Reply:
Makes sense, thanks.

Reply:
@dcapwell I think this is always an `ArrayList`, so simplifying is probably not going to hurt us, but I'll leave the decision to you.

Reply:
For what little it is worth, I would like to see this become a straightforward norm in the codebase, to ensure we are not paying unnecessary costs - and it bears no clarity penalty.  But it's early days for instituting that norm, so no pressure here, just mentioning it.

Reply:
I'm fine with that. Although if we do it, I'd prefer something like...

```
int size = fieldNames.size();
for (int i = 0; i < size; i++)
```

...for clarity.

Reply:
I personally find it much clearer to use 
`for (int i = 0, size = fieldNames.size() ; i < size ; ++i)`

Note that this binds the scope of size to the for loop, which is particularly helpful for maintaining norms across a code block with multiple loops, and also is more typical style.

Reply:
I also prefer `for (int i = 0, size = fieldNames.size() ; i < size ; ++i)` as it limits the scope of size.

Reply:
WFM, as long as it's not `m` :p